x-pack/plugins/security_solution/public/management/components/hooks/use_url_pagination.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { MANAGEMENT_PAGE_SIZE_OPTIONS } from '../../common/constants';
+import { useUrlParams } from './use_url_params';
+
+// Page is not index-based, it is 1-based
+interface Pagination {
+  page: number;
+  pageSize: number;
+}
+
+type SetUrlPagination = (pagination: Pagination) => void;
+interface UrlPagination {
+  pagination: Pagination;
+  setPagination: SetUrlPagination;
+  pageSizeOptions: number[];
+}
+
+type UrlPaginationParams = Partial<Pagination>;
+
+const paginationFromUrlParams = (urlParams: UrlPaginationParams): Pagination => {
+  const pagination: Pagination = {
+    pageSize: 20,
+    page: 1,
+  };
+
+  // Search params can appear multiple times in the URL, in which case the value for them,
+  // once parsed, would be an array. In these case, we take the last value defined
+  pagination.page = Number(
+    (Array.isArray(urlParams.page) ? urlParams.page.pop() : urlParams.page) ?? pagination.page
```

Comment:
Would this condition ever be true `Array.isArray(urlParams.page)` since page is always either `number` or `undefined`? Likewise for `pageSize`.

Reply:
yes it will be true when multiple values are defined because that's what is returned from `parse(search)` in the `useUrlParams` hook